packages/neo-one-client-core/src/provider/NEOONEOneDataProvider.ts

```diff
@@ -235,7 +238,7 @@ export class NEOONEOneDataProvider implements DeveloperProvider {
   private async getProvider(): Promise<NEOONEDataProvider> {
     /* istanbul ignore next */
     if (this.mutableProvider === undefined) {
-      const client = new OneClient(this.port);
+      const client = new OneClient(this.port, this.host);
```

Comment:
Use `this.host` below to modify the `rpcURL`, something like `result.response.nodes[0].rpcAddress.replace('localhost', this.host)`